avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_4/UNION_GenericDeserializer_3052180383482420701_4807922374121811220.java

```diff
@@ -0,0 +1,69 @@
+
+package com.linkedin.avro.fastserde.generated.deserialization.AVRO_1_4;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import com.linkedin.avro.fastserde.FastDeserializer;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.util.Utf8;
+
+public class UNION_GenericDeserializer_3052180383482420701_4807922374121811220
+    implements FastDeserializer<Map<Utf8, IndexedRecord>>
+{
+
+    private final Schema readerSchema;
+    private final Schema mapMapSchema0;
+    private final Schema mapMapValueSchema0;
+
+    public UNION_GenericDeserializer_3052180383482420701_4807922374121811220(Schema readerSchema) {
+        this.readerSchema = readerSchema;
+        this.mapMapSchema0 = readerSchema.getTypes().get(1);
+        this.mapMapValueSchema0 = mapMapSchema0 .getValueType();
+    }
+
+    public Map<Utf8, IndexedRecord> deserialize(Map<Utf8, IndexedRecord> reuse, Decoder decoder)
+        throws IOException
+    {
+        Map<Utf8, IndexedRecord> map0 = null;
+        long chunkLen0 = (decoder.readMapStart());
+        if (chunkLen0 > 0) {
+            Map<Utf8, IndexedRecord> mapReuse0 = null;
+            if ((reuse) instanceof Map) {
+                mapReuse0 = ((Map)(reuse));
+            }
+            if (mapReuse0 != (null)) {
+                mapReuse0 .clear();
+                map0 = mapReuse0;
+            } else {
+                map0 = new HashMap<Utf8, IndexedRecord>(((int)(((chunkLen0 * 4)+ 2)/ 3)));
+            }
+            do {
+                for (int counter0 = 0; (counter0 <chunkLen0); counter0 ++) {
```

Comment:
nit - code generation should emit a space after the < operator

Reply:
Unfortunately, this seems to be the quirk of the codemodel library:

```java
    JDoLoop doLoop = ifBlockForChunkLenCheck._do(chunkLen.gt(JExpr.lit(0)));
    JForLoop forLoop = doLoop.body()._for();
    JVar counter = forLoop.init(codeModel.INT, getUniqueName("counter"), JExpr.lit(0));
    forLoop.test(counter.lt(chunkLen));
    forLoop.update(counter.incr());

```

Reply:
:-(